### PR TITLE
MAW-598: Added callback for custom response handlers

### PR DESCRIPTION
**Pull request by ghost**
**Is there any context or background the reviewer should know about?**
In process of working with `two-legged-auth` library, I faced with necessity to handle payloads with non-200 response codes. For example, chapstick venue API returns duplicate entities with 409 code (Conflicts). 

**What does this PR do?**
Adds callback for custom response handlers 

**Where should the reviewer start with this PR?**
...

**How did I test this PR?**
Run unit tests

**Checklist**
- [X] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] I have specified all launch tasks in the JIRA ticket

**PR Policy for this repository**

Add the label `core-services-ready` to your PR when it is ready for review by core services. Please have your PR reviewed by someone on your team before adding this label to your PR. **In case of emergency** please request a review in #core-services-ask (slack). Otherwise, SLA for review is 24 hours (adjust for non-holiday office hours schedule, US/EST). For more details refer to https://meetup.atlassian.net/wiki/x/WwBvEw.

### CODE REVIEW

File: src/main/scala/com/meetup/auth/MeetupConsumer.scala [hunk 1]
```diff
@@ -116,23 +121,26 @@ class MeetupConsumerImpl(configuration: Configuration)(
...
-        log.error(
-          s"received non-200 response from POST: $hostAndPath\n" +
-            s"with contentBody:\n    $contentBody\n" +
-            s"and headers:\n    $headers\n" +
-            s"and http status code: ${r.getStatusCode}\n" +
-            s"and body: ${r.getResponseBody}\n"
-        )
+        log.error(s"received non-200 response from $methodType\n" +
+          s"and http status code: ${r.getStatusCode}\n" +
+          s"and body: ${r.getResponseBody}\n")
```
Comment (mkleymenov) [inline src/main/scala/com/meetup/auth/MeetupConsumer.scala]: Could you please remove line breaks from the log? It causes log aggregators to record a single log statement as multiple log records

Comment (mkleymenov) [reply src/main/scala/com/meetup/auth/MeetupConsumer.scala]: Also you don't log the request url and payload. I think we should log at least the request URL to be able to correlate errors with requests

**Review by mkleymenov — APPROVED** [review]
LGTM

File: src/main/scala/com/meetup/auth/MeetupConsumer.scala (hunk 1 above)
Comment (ghost) [reply src/main/scala/com/meetup/auth/MeetupConsumer.scala]: Done.

**Pull request merged**